src/dc_msg.rs

```diff
@@ -472,7 +472,17 @@ pub fn dc_msg_load_from_db<'a>(msg: *mut dc_msg_t<'a>, context: &'a Context, id:
                 (*msg).type_0 = row.get(12)?;
                 (*msg).state = row.get(13)?;
                 (*msg).is_dc_message = row.get(14)?;
-                (*msg).text = row.get::<_, Option<String>>(15)?;
+                (*msg).text = Some(if let rusqlite::types::ValueRef::Text(buf) = row.get_raw(15) {
+                    if let Ok(t) = String::from_utf8(buf.to_vec()) {
+                        t
+                    } else {
+                        warn!(context, 0, "dc_msg_load_from_db: could not get text column as non-lossy utf8 id {}", id);
+                        String::from_utf8_lossy(buf).into_owned()
+                    }
+                } else {
+                    warn!(context, 0, "dc_msg_load_from_db: could not get text column for id {}", id);
+                    "[ Could not read from db ]".to_string()
+                });
```

Comment:
I don't find this whole construct the most readable.  Maybe a temporary variable is easier, at least it limits hunting for the closing Some bracket.  Maybe match is easier too.  A lot of this is probably all taste (I'm in general not a huge if let fan anyway).  So entirely up to you what you do with this comment really ;)

Reply:
Yup you're right, changed it to what you recommended. "Returning" if let statements aren't the most readable if they get longer.